src/compiler.ts

```diff
@@ -462,6 +462,36 @@ export class Compiler extends DiagnosticEmitter {
     return module;
   }
 
+  addNullPointerExceptionHandling(module: Module) {
+    if (this.options.hasFeature(Feature.EXCEPTION_HANDLING)) {
+      let message = this.ensureStaticString("null is not a function");
+      let exceptionType = module.addFunctionType(
+        "NullPointerException",
+        NativeType.None,
+        [NativeType.I32]
+      );
+
+      module.addEvent("NullPointerException", 0, exceptionType);
```

Comment:
For that real Java feels

Reply:
btw this string is only used as a .wat debugging symbol it isn't used at runtime. Wasm exceptions don't (currently) have names like JS exceptions do. They're all `WebAssembly.Runtime` instances with a message field of just `wasm exception` from the JS side of things.